tasks/gen/resource/browser.cr

```diff
@@ -0,0 +1,194 @@
+    validate_has_valid_columns!
+  end
+
+  private def validate_name_is_present!
+    if resource_name?.nil? || resource_name?.try &.empty?
+      error "Resource name is required. Example: lucky gen.resource.browser User"
+    end
+  end
+
+  private def validate_not_namespaced!
+    if resource_name.includes?("::")
+      error "Namespaced resources are not supported"
+    end
+  end
+
+  private def validate_name_is_singular!
+    singularized_name = LuckyInflector::Inflector.singularize(resource_name)
+    if singularized_name != resource_name
+      error "Resource must be singular. Example: lucky gen.resource.browser #{singularized_name}"
+    end
+  end
+
+  private def validate_name_is_camelcase!
+    if resource_name.camelcase != resource_name
+      error "Resource name should be camel case. Example: lucky gen.resource.browser #{resource_name.camelcase}"
+    end
+  end
+
+  private def validate_has_valid_columns!
+    if !columns_are_valid?
+      error "Must provide valid columns for the resource: lucky gen.resource.browser #{resource_name.camelcase} name:String"
+    end
+  end
+
+  private def error(message : String)
+    raise InvalidOption.new(message)
+  end
+
+  private def column_definitions
+    if column_arguments?
+      ARGV.skip(1)
+    else
+      [] of String
+    end
+  end
+
+  private def column_arguments? : Bool
+    !!ARGV[1]?
+  end
+
+  private def columns_are_valid? : Bool
+    column_definitions.any? && column_definitions.all? do |column_definition|
+      column_parts = column_definition.split(":")
+      column_name = column_parts.first
+      column_parts.size == 2 && column_name == column_name.underscore
+    end
+  end
+
+  private def display_success_messages
+    success_message(resource_name, "./src/models/#{underscored_resource}.cr")
+    success_message(resource_name + "Form", "./src/forms/#{underscored_resource}_form.cr")
+    success_message(resource_name + "Query", "./src/queries/#{underscored_resource}_query.cr")
+    %w(index show new create edit update delete).each do |action|
+      success_message(
+        pluralized_resource + "::" + action.capitalize,
+        "./src/actions/#{folder_name}/#{action}.cr"
+      )
+    end
+    %w(index show new edit).each do |action|
+      success_message(
+        pluralized_resource + "::" + action.capitalize + "Page",
+        "./src/pages/#{folder_name}/#{action}_page.cr"
+      )
+    end
+  end
+
+  private def underscored_resource
+    resource_name.underscore
+  end
+
+  private def folder_name
+    LuckyInflector::Inflector.pluralize underscored_resource
+  end
+
+  private def pluralized_resource
+    LuckyInflector::Inflector.pluralize resource_name
+  end
+
+  private def success_message(class_name : String, filename : String) : Void
+    io.puts "Generated #{class_name.colorize.green} in #{filename.colorize.green}"
+  end
+
+  private def resource_name
+    resource_name?.not_nil!
+  end
+
+  private def resource_name?
+    ARGV.first?
+  end
```

Comment:
I'm not loving how these generators work. In the future I want to extract something to consolidate a lot of the common stuff and make it simpler (things like log output, validations, etc)